### ISSUE TITLE
bw login returning malformed json with some responses

### ISSUE THREAD

**Issue by ggiesen** [bitwarden/clients#4521]
### Steps To Reproduce

The bw `login` function is returning malformed JSON:

login (successful):
```
$ /usr/local/bin/bw login user@example.com --apikey --response
{"success":true,"data":{"noColor":false,"object":"message","title":"You are logged in!","message":"\nTo unlock your vault, use the `unlock` command. ex:\n$ bw unlock"}}
```

Note the `\n` that appear.

login (unsuccessful):
```
/usr/local/bin/bw login user@example.com --apikey --response
{"success":false,"message":"{\"response\":{\"error\":\"invalid_client\"},\"captchaRequired\":false,\"statusCode\":400}"}
```

Note the escaped quotes (`\"`)

### Expected Result

login (successful):
```
$ /usr/local/bin/bw login user@example.com --apikey --response
{"success":true,"data":{"noColor":false,"object":"message","title":"You are logged in!","message":"To unlock your vault, use the `unlock` command. ex:$ bw unlock"}}
```

login (unsuccessful)
```
$ /usr/local/bin/bw login user@example.com --apikey --response
{"success":false,"message":{"response":{"error":"invalid_client"},"captchaRequired":false,"statusCode":400}}
```

### Actual Result

login (successful):
```
$ /usr/local/bin/bw login user@example.com --apikey --response
{"success":true,"data":{"noColor":false,"object":"message","title":"You are logged in!","message":"\nTo unlock your vault, use the `unlock` command. ex:\n$ bw unlock"}}
```

Note the `\n` that appear.

login (unsuccessful):
```
/usr/local/bin/bw login user@example.com --apikey --response
{"success":false,"message":"{\"response\":{\"error\":\"invalid_client\"},\"captchaRequired\":false,\"statusCode\":400}"}
```

Note the escaped quotes (`\"`)

### Operating System

Linux

### Operating System Version

Rocky Linux release 8.5 (Green Obsidian)

### Shell

Bash

### Build Version

2023.1.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Krychaz** [comment]
Hello there,

I was not able to reproduce this issue, however I will see if QA is able to do it

**Comment by joshtrichards** [comment]
I can reproduce this issue, but technically that *is* valid JSON. The literal output to the user is what's provided in the `message` field when `--response` is specified. And it'll nearly always need some escaping. 

It just happens that - in the case of an unsuccessful login - JSON formatted output is *always* provided to the user **(even without the `--response` command being specified)**. That's in contrast to what happens by default when things are successful. 

In the case of unsuccessful logins, if `--response` is specified this happens to result in the original JSON "message" being encoded so it can be embedded as a string in the `message` field. At that point it's just a string that still happens to hint at its JSON origins.

This could perhaps be handled a bit tidier, but there's technically nothing wrong there that I can see.

I think the real issue is that the CLI is inconsistent as to when it does/doesn't provide JSON output even before the `---response` parameter even comes into play. Specifying `--response` just makes the issue more obvious.

Real examples should help clarify.

## Here's the output when logging in successfully without `--response`:

```
~$ ./bw logout && BW_CLIENTID=user.[a valid id] BW_CLIENTSECRET=[a valid secret] ./bw login --apikey
You are logged in!

To unlock your vault, use the `unlock` command. ex:
$ bw unlock
```
Looks good!

## Here's the output during a successful login when `--response` is specified:

(I appended `--pretty` to make it easier read otherwise it's the same as @ggiesen provided):

```
~$ ./bw logout && BW_CLIENTID=user.[a valid id] BW_CLIENTSECRET=[a valid secret] ./bw login --apikey --response --pretty
{
  "success": true,
  "data": {
    "noColor": false,
    "object": "message",
    "title": "You are logged in!",
    "message": "\nTo unlock your vault, use the `unlock` command. ex:\n$ bw unlock"
  }
}
```

Looks good!

## Here's the output when the login attempt fails without `--response`:

```
~$ ./bw logout && BW_CLIENTID=user.[a bogus id] BW_CLIENTSECRET=[an invalid secret] ./bw login --apikey --pretty
{"response":{"error":"invalid_client"},"captchaRequired":false,"statusCode":400}
```
Why'd we get JSON here by default? That's odd (at least in comparison to behavior during successful logins).

Note, again, this is *without* specifying `--response`.

(Incidentally specifying `--pretty` seems to be no-op here - doesn't matter whether I specify it or not and - perhaps another indication that JSON output by default wasn't really a deliberate choice here, but that's a WAG on my part). To be consistent I'll run the above unexpected response through `jq`:

```
{
  "response": {
    "error": "invalid_client"
  },
  "captchaRequired": false,
  "statusCode": 400
}
```

I don't see a real issue with this. Theoretically successful logins could even be indicated without even adjusting the structure - just returning `false` as the value for the `error` field. But, again, successful logins don't even return any JSON so still odd/inconsistent.

## Here's the output when the login attempt fails when `--response` is specified:

```
~$ ./bw logout && BW_CLIENTID=user.[a bogus id] BW_CLIENTSECRET=[an invalid secret] ./bw login --apikey --response --pretty
{
  "success": false,
  "message": "{\"response\":{\"error\":\"invalid_client\"},\"captchaRequired\":false,\"statusCode\":400}"
}
```

Since the original (i.e. non-`--response` output) already included a JSON response in stdout to the user, the impact of specifying `--response` was just the addition of the `success` field (and embedding the original already JSON output as a more inconvient string).

## Analysis

1. The `message` field (and `title` but same diff) *is* always the literal output string sent to the user on stdout so they're encoded strings. There will always be escaping when/if necessary. **No issue there.**
2. It happens that whenever the login is successful that some additional fields are included in another object (`data`) in which `message` gets stashed, but that's sort of a distraction from the real issue IMO. **No issue there.**
3. Login failures result in JSON output by default regardless of command line parameters. While arguably unfriendly to an interactive user it's not like it's the only time JSON is spit out by the CLI given most vault actions are JSON based, but this behavior is inconsistent with the behavior for successful logins (which don't output any JSON at all). **This is a cosmetic issue at best, and, at worst, a precursor to oddities that must be handled by the user in odd ways when `--response` gets specified**
4. The `success` field is maybe unnecessary (regardless of success or failure of the login)
5. The JSON output during failures is helpful. There isn't any during successes.

Additional questions:

1. Specifying `--response` serves what purpose? Does it only exist because of the inconsistent behavior between successful and failed login attempts?

Ideas:

1. Make the output to the user during failures more consistent to successful ones (e.g. including what will later become a `title` and `message` if JSON output is generated but otherwise simply be, well, a title and message).
2. Either always (success or failure) output a JSON response by default or to never (success or failure ) output a JSON response - unless `--response` is specified
3. Either nuke the `success` field (since it kind of is redundant when/if successful and failed logins are handled consistently) or handle the rest of the JSON fields that failures generate more cleanly (i.e. embed them alongside the rest of the fields in in JSON rather than as a string)

This, as a side effect, would eliminate the inconvenient JSON-in-JSON encoding that @ggiesen ran across. The message field will always be encoded, but that's to be expected and how it should be.

**Comment by joshtrichards** [comment]
Tldr: 

- The JSON output provided by `--response` is valid albeit arguably suboptimal (at least during the login portion of workflows)
- Sometimes, by default, JSON output is provided and sometimes it isn't during various login scenarios and this seems inadvertent 
- Inconsistent response data handling before `--response` is even specified seems to result in suboptimally structured JSON response output during different login scenarios and complicates usage of it by users
- Making JSON appear consistently/cohesively regardless of login scenarios and whether or not `--response` is specified would seem friendlier and probably the intention while simultaneously eliminating the inconvenient/unnecessary encoding of `message` JSON in an embedded string 
- Some specific ideas are left for further discussion (and explored some in the above write-up in more depth)

**Comment by ggiesen** [comment]
Yes perhaps I should restate that it's not invalid, but inconsistent. I'm writing an [extension](https://gitlab.com/ggiesen/salt-ext-bitwarden) for [Salt](https://saltproject.io) and while primarily using the [Vault Management API](https://bitwarden.com/help/vault-management-api/), I must use the CLI for login/logout. The login/logout functions' output behave differently than any other function, whether it's the CLI or the REST API, and I was having to treat it as a special case for both my code and my tests.
